internal/credentials/model.go

```diff
@@ -81,6 +84,14 @@ func (m *Model) GetPassword() string {
 	return string(pwd)
 }
 
+// GetHiddenPassword returns the Password after
+// masking all but last 4 characters with *
+func (m *Model) GetHiddenPassword() string {
+	// Returns a hidden password which contains a string of asterisk
+	// followed by last 4 characters of the original password
+	return AsteriskString + m.GetPassword()[len(m.GetPassword())-4:len(m.GetPassword())]
```

Comment:
call `m.GetPassword()` once and store it in a variable. use the variable for subsequent index calculation/length calculation.

every call to `m.GetPassword()` decrypts from hex.

Reply:
Also, check if the length of `m.GetPassword` is > 4 ? probably there is no chance that the password is shorter than a 4 letter character.. At least mention this as a warning?

Reply:
I think we should add some validations in `func NewCredential(username, password string) *Model`  to make sure valid inputs are sent. For now, only len of password can be checked, but it can be extended in future if needed. There is a `IsValidUsername` which can be called. Thoughts?

Reply:
Validation may not be required now since we are generating the password and not taking an input.

Reply:
Okay. Adding a warning in `type Model struct` to mention that password len must be > 4